src/window.js

```diff
@@ -1,677 +1,103 @@
+        onFocusGlobalSearch: this.onFocusGlobalSearch,
+      });
     }
-  });
-
-  webview.get_back_forward_list().connect("changed", () => {
-    updateButtons();
-  });
-
-  function updateButtons() {
-    button_back.sensitive = webview.can_go_back();
-    button_forward.sensitive = webview.can_go_forward();
-  }
-
-  button_back.connect("clicked", () => {
-    webview.go_back();
-  });
-
-  button_forward.connect("clicked", () => {
-    webview.go_forward();
-  });
 
-  const adj = browse_page.get_vscrollbar().adjustment;
-  adj.connect("value-changed", () => {
-    if (scrolled_to) {
-      const index = browse_selection_model.selected;
-      const bottom_edge = (index + 1) * 38 - adj.value;
-      const top_edge = bottom_edge - 38;
-      // If row is not visible after scroll_to, adjust
-      if (bottom_edge === 0) {
-        adj.value -= 38;
-      } else if (top_edge === adj.page_size) {
-        adj.value += 38;
-      }
-      scrolled_to = false;
+    #createSidebar() {
+      this._sidebar = new Sidebar({ webview: this._webview });
+      this._split_view.sidebar = this._sidebar;
     }
-  });
 
-  function onSearchChanged() {
-    if (search_entry.text) {
-      stack.visible_child = search_page;
-      const selection_model = search_list_view.model;
-      selection_model.unselect_item(selection_model.selected);
-      selection_model.model.model.filter = createFilter(search_entry.text);
-      if (!selection_model.n_items) stack.visible_child = status_page;
-      search_list_view.scroll_to(0, Gtk.ListScrollFlags.NONE, null);
-    } else {
-      stack.visible_child = browse_page;
+    #addThemeSelector() {
+      const button_menu = this._sidebar._button_menu;
+      const popover = button_menu.get_popover();
+      popover.add_child(new ThemeSelector(), "themeswitcher");
     }
-  }
 
-  search_entry.connect("search-changed", onSearchChanged);
+    #connectWebView() {
+      this._webview.connect("load-changed", () => {
+        this.#updateButtons();
+      });
 
-  let promise_load;
-  async function load() {
-    if (!promise_load) {
-      promise_load = Promise.all([
-        scanLibraries(root_model, Gio.File.new_for_path("/usr/share/doc")),
-        scanLibraries(
-          root_model,
-          Gio.File.new_for_path("/usr/share/gtk-doc/html"),
-        ),
-        scanLibraries(root_model, Gio.File.new_for_path("/app/share/doc")),
-      ]).then(() => {
-        const search_model = flattenModel(root_model);
-        search_list_view.model = createSearchSelectionModel(
-          search_model,
-          browse_list_view,
-        );
+      this._webview.get_back_forward_list().connect("changed", () => {
+        this.#updateButtons();
       });
     }
-    return promise_load;
-  }
-
-  function present() {
-    // The window is already open
-    const mapped = window.get_mapped();
-    window.present();
-    onFocusGlobalSearch();
-    load()
-      .then(() => {
-        if (!mapped) {
```

Comment:
None of this stuff should be needed anymore as it was required when Manuals was hidden and re-opened in Workbench.